lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -233,13 +234,25 @@ class Chef::Knife::VsphereVmClone < Chef::Knife::BaseVsphereCommand
          description: 'Indicates whether to mark the new vm as a template',
          boolean: false
 
+  option :random_vmware,
+         long: '--random-vmname',
+         description: 'Creates a random VMNAME starts with vm-XXXXXXXX',
+         boolean: false
+
+  option :random_vmware_prefix,
+         long: '--random-vmname-prefix PREFIX',
+         description: 'Change the VMNAME prefix',
+         default: 'vm-'
+
   def run
     $stdout.sync = true
 
     vmname = @name_args[0]
-    if vmname.nil?
+    if (vmname.nil? && !config[:random_vmware]) || (vmname && config[:random_vmware])
```

Comment:
What would you think about some private methods to better communicate the intent of this section? There are two conditions, and we only want one. If zero or two are set then that's an error. I was thinking something along the lines of:

```Ruby
if using_supplied_hostname? ^ using_random_hostname?
  vmname = supplied_hostname || random_hostname
else
  show_usage
  fatal_exit('You must specify a virtual machine name OR use --random-vmname')
end

private

def using_random_hostname?
  config[:random_vmware]
end

def using_supplied_hostname?
  ! supplied_hostname.nil?
end

def supplied_hostname
  @name_args[0]
end

def random_hostname
  config[:random_vmware_prefix] + SecureRandom.hex(4)
end
```

Now that I'm typing this out we don't even need to assign `vmname`, it can be a private method and all we need to do is check for the error condition:

```Ruby
unless using_supplied_hostname? ^ using_random_hostname?
  show_usage
  fatal_exit('You must specify a virtual machine name OR use --random-vmname')
end

private

def vmname
  supplied_hostname || random_hostname
end

def using_random_hostname?
  config[:random_vmware]
end

def using_supplied_hostname?
  ! supplied_hostname.nil?
end

def supplied_hostname
  @name_args[0]
end

def random_hostname
  @random_hostname ||= config[:random_vmware_prefix] + SecureRandom.hex(4)
end
```